phoenix-core/src/main/java/org/apache/phoenix/schema/TableRef.java

```diff
@@ -101,7 +105,7 @@ else if (SchemaUtil.isPKColumn(column)) {
     public int hashCode() {
         final int prime = 31;
         int result = alias == null ? 0 : alias.hashCode();
-        result = prime * result + this.table.getName().getString().hashCode();
+        result = prime * result + ( this.table!=null && this.table.getName()!=null ? this.table.getName().getString().hashCode() : 0);
```

Comment:
Can the table in a TableRef be null? If not, then you probably don't need the null check. If it cannot be null, then it would be also good to add Preconditions.checkNotNull() checks in various constructors of this class. If it can be null, then you also need to modify the equals method too (commented below).